### PR TITLE
prechecks: add prechecks role

### PR DESCRIPTION
**Pull request by leseb**
This role contains all the necessary checks for a Ceph cluster to be
deployed. Instead of running the checks in each individual role, having
a central role is easier to manage. It's also easier to play at the very
beginning of the site.yml

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
rebase on top of https://github.com/ceph/ceph-ansible/pull/2036

**Comment by leseb** [comment]
todo, add  `serial: '{{ serial|default("1") }}'` to `rolling_update.yml`

File: site-docker.yml.sample [hunk 1]
```diff
@@ -15,12 +15,16 @@
...
+  vars:
+    delegate_facts_host: True
   pre_tasks:
     - name: gather and delegate facts
       setup:
       delegate_to: "{{ item }}"
       delegate_facts: True
       with_items: "{{ groups['all'] }}"
+      tags: always
+      when: delegate_facts_host | bool
```
Comment (leseb) [inline site-docker.yml.sample]: add the opposite too when delegate_facts_host is false we still need to gather facts

**Comment by guits** [comment]
closing this since ceph-validate has been merged

**Pull request closed without merging**